app/models/course/material/folder.rb

```diff
@@ -6,6 +6,7 @@ class Course::Material::Folder < ActiveRecord::Base
 
   has_many :materials, inverse_of: :folder, dependent: :destroy, foreign_key: :folder_id,
                        class_name: Course::Material.name, autosave: true
+  has_many :downloads, inverse_of: :folder, dependent: :destroy
```

Comment:
`course/:id/material/folder/:id/downloads/:id`, will this url make sense ?  or maybe `course/:id/material/folder/:id/download_job/:id` ?